test/functional/getblockstats.py

```diff
@@ -0,0 +1,186 @@
+        assert_equal(stats[1]['minfeerate'], 20)
+        assert_equal(stats[1]['maxfeerate'], 20)
+        assert_equal(stats[1]['medianfeerate'], 20)
+        assert_equal(stats[1]['avgfeerate'], 20)
+        # assert_equal(stats[1]['mintxsize'], 192)
+        # assert_equal(stats[1]['maxtxsize'], 192)
+        # assert_equal(stats[1]['mediantxsize'], 192)
+        # assert_equal(stats[1]['avgtxsize'], 192)
+
+        assert_equal(stats[max_stat_pos]['txs'], 4)
+        assert_equal(stats[max_stat_pos]['swtxs'], 0)
+        assert_equal(stats[max_stat_pos]['ins'], 3)
+        assert_equal(stats[max_stat_pos]['outs'], 8)
+        assert_equal(stats[max_stat_pos]['totalfee'], 76160)
+        assert_equal(stats[max_stat_pos]['utxo_increase'], 5)
+        assert_equal(stats[max_stat_pos]['utxo_size_inc'], 388)
+        # assert_equal(stats[max_stat_pos]['total_size'], 643)
+        # assert_equal(stats[max_stat_pos]['total_weight'], 2572)
+        assert_equal(stats[max_stat_pos]['total_out'], 9999920000)
+        assert_equal(stats[max_stat_pos]['minfee'], 3840)
+        assert_equal(stats[max_stat_pos]['maxfee'], 67800)
+        assert_equal(stats[max_stat_pos]['medianfee'], 4520)
+        assert_equal(stats[max_stat_pos]['avgfee'], 25386)
+        assert_equal(stats[max_stat_pos]['minfeerate'], 20)
+        # assert_equal(stats[max_stat_pos]['maxfeerate'], 300)
+        assert_equal(stats[max_stat_pos]['medianfeerate'], 20)
+        assert_equal(stats[max_stat_pos]['avgfeerate'], 118)
+        # assert_equal(stats[max_stat_pos]['mintxsize'], 192)
+        # assert_equal(stats[max_stat_pos]['maxtxsize'], 226)
+        # assert_equal(stats[max_stat_pos]['mediantxsize'], 225)
+        # assert_equal(stats[max_stat_pos]['avgtxsize'], 214)
+
+        # Test invalid parameters raise the proper json exceptions
+        tip = start_height + max_stat_pos
+        assert_raises_rpc_error(-8, 'Target block height %d after current tip %d' % (tip+1, tip), node.getblockstats, height=tip+1)
+        assert_raises_rpc_error(-8, 'Target block height %d after current tip %d' % (-1, tip), node.getblockstats, height=-tip-1)
+
+        # Make sure not valid stats aren't allowed
+        inv_sel_stat = 'asdfghjkl'
+        inv_stats = [
+            'minfee,%s' % inv_sel_stat,
+            '%s,minfee' % inv_sel_stat,
+            'minfee,%s,maxfee' % inv_sel_stat,
+        ]
+        for inv_stat in inv_stats:
+            assert_raises_rpc_error(-8, 'Invalid selected statistic %s' % inv_sel_stat, node.getblockstats, height=1, stats=inv_stat)
+        # Make sure we aren't always returning inv_sel_stat as the culprit stat
+        assert_raises_rpc_error(-8, 'Invalid selected statistic aaa%s' % inv_sel_stat, node.getblockstats, height=1, stats='minfee,aaa%s' % inv_sel_stat)
+
+        # Make sure only the selected statistics are included
+        stats = node.getblockstats(height=1, stats='minfee,maxfee')
+        some_values = [
```

Comment:
In commit "QA: Test new getblockstats RPC"

Again, seems like it would be simpler to just check for equality:

```python
stats = node.getblockstats(height=1, stats='minfee,maxfee')
assert_equal(set(stats), {"minfee", "maxfee"})
```